Autopilot/DataStructures/Src/vectorClass.cpp

```diff
@@ -0,0 +1,108 @@
+/* 
+    Author: Nixon Chan
+*/
+
+#include <math.h>
+#include "vectorClass.hpp"
+
+Vector3D::Vector3D()
+{
+    x = 0;
+    y = 0;
+    z = 0;
+}
+
+Vector3D::Vector3D(double inputX, double inputY, double inputZ)
+{
+    x = inputX;
+    y = inputY;
+    z = inputZ;
+}
+
+Vector3D::~Vector3D()
+{
+}
+
+double Vector3D::getX() const
+{
+    return x;
+}
+
+double Vector3D::getY() const
+{
+    return y;
+}
+
+double Vector3D::getZ() const
+{
+    return z;
+}
+
+void Vector3D::setX(double input)
+{
+    x = input;
+}
+
+void Vector3D::setY(double input)
+{
+    y = input;
+}
+
+void Vector3D::setZ(double input)
+{
+    z = input;
+}
+
+Vector3D Vector3D::operator=(const Vector3D &vector2)
+{
+    (*this).x = vector2.x;
```

Comment:
The good people who make c++ created arrow notation specifically to avoid the need to worry about brackets around pointer de referencing when accessing members.

(*this).x    is equivalent to      this->x

Reply:
Fixed!